### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by Suedie**
Added tests and a pub-sub pattern for sound

Sounds are a bit loud

### CODE REVIEW

File: core/src/model/Model.java [hunk 1]
```diff
@@ -25,8 +27,10 @@ public class Model {
     private final RewardSystem rewardSystem = new RewardSystem();
     private final World<IEntity> world;
     private int currentScore = 0;
+    private final List<ISoundSubscriber> soundSubscribers = new ArrayList<>();
+    private boolean playerIsDead = false;
```
Comment (pkasinski) [inline core/src/model/Model.java:31]: Have you considered putting this boolean in the class Player? Is it even needed?

File: core/src/model/IModelPublisher.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package model;
+
+import view.ISoundSubscriber;
+
+/**
+ * Allows the model to publish to any subscriber that changes
+ * in the model have occured.
+ */
+public interface IModelPublisher {
```
Comment (rojtekin) [inline core/src/model/IModelPublisher.java]: Shouldnt this be namned IModelObserver?

**Review by pkasinski — APPROVED** [review]
Looks good. I left a comment.

File: core/src/model/IWeaponPublisher.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package model;
+
+import view.ISoundSubscriber;
+
+/**
+ * Allows the weapon to publish to any subscriber that a swordswing has occurred
+ */
+public interface IWeaponPublisher {
```
Comment (rojtekin) [inline core/src/model/IWeaponPublisher.java]: why publisher and not observer?

File: core/src/view/ISoundSubscriber.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package view;
+
+public interface ISoundSubscriber {
```
Comment (rojtekin) [inline core/src/view/ISoundSubscriber.java]: observer over publisher?

File: core/src/view/Sound.java [hunk 5]
```diff
@@ -34,13 +34,13 @@ public void stopGameMusic(){
      * Plays the sound associated with the enemy hitting something/someone.
      */
     public void playEnemyHit(){
-        ENEMY_HIT.play(); // TODO: check if the sounds isnt too loud later
+        ENEMY_HIT.play(); // TODO: check if the sounds isn't too loud later
```
Comment (rojtekin) [inline core/src/view/Sound.java]: to reduce the volume put a float inside () between 1 and 0. where 1 is full volume

File: core/src/view/Sound.java [hunk 6]
```diff
@@ -34,13 +34,13 @@ public void stopGameMusic(){
...
     }
 
     /**
      * Plays the sound of the sword swing.
      */
-    public  void playSwordHit(){
+    public  void playSwordSwing(){
         SWORD_SWOOSH.play(); // TODO: check if the sounds isnt too loud later
```
Comment (rojtekin) [inline core/src/view/Sound.java]: same as above

**Review by rojtekin — APPROVED** [review]
LGTM, just make sure the volume of swordhit and death isnt too loud

File: core/src/model/IModelPublisher.java (hunk 2 above)
Comment (Suedie) [reply core/src/model/IModelPublisher.java]: You are right, fixing this

**Pull request merged**